pkg/reconciler/pipeline/controller.go

```diff
@@ -734,6 +749,19 @@ func (r *pipelineReconciler) resumePipeline(ctx context.Context, pl *dfv1.Pipeli
 }
 
 func (r *pipelineReconciler) pausePipeline(ctx context.Context, pl *dfv1.Pipeline) (bool, error) {
+
+	if pl.GetAnnotations() == nil || pl.GetAnnotations()[dfv1.KeyPauseTimestamp] == "" {
```

Comment:
No else?

Reply:
For this else, we can just return `false` since it means there is a timestamp set.